test/service-stream.test.js

```diff
@@ -0,0 +1,190 @@
+/*global threads,assert,suite,setup,teardown,test*/
+/*jshint esnext:true, maxlen:false*/
+
+suite('Stream /', function() {
+  suite('ServiceStream /', function() {
+
+    // we use these tests basically just to check edge cases, regular use cases
+    // are handled by the client tests
+    var service;
+    var lastMessage;
+    var activeStream;
+    var STREAM_ID = 'fake-stream-id';
+
+    setup(function() {
+      service = threads.service('test-stream');
+    });
+
+    function callStream(methodName, fn) {
+      service.stream(methodName, fn);
+      service.private.channels['123'] = {
```

Comment:
it would if I could access the `ServiceStream` constructor -- which is a private constructor at the moment and our test infrastructure don't really load the src files, just the built file.